User Management for Admins

Zukye opened the pull request:
user management for admins (add, remove, edit...) + language selection issue fix

Commit:
usermanagement wip

Commit:
UserDataObjekt

Commit:
edit user functionality for admins

Commit:
get user config by id

Commit:
Added check for empty password

DivineThreepwood committed:
replaced german special character

(cherry picked from commit 35a432e)

Commit:
W.I.P

(cherry picked from commit 10c0dd776c22ee71c98d4e877befe132222d3e78)

Commit:
Merge branch 'master' into temp-user_management

Commit:
Add Error-messages

Commit:
Change Tests for SesseionMaagerFacade

Commit:
remove Password from UserData

Commit:
Disable password-fields on user-editing, cleanup

Commit:
Fix verifying of username

Commit:
user feedback for user management

DivineThreepwood commented on src/main/resources/languages/languages_de_DE.properties:
```diff
@@ -31,7 +31,7 @@ BRIGHTNESS_SENSOR=Helligkeitssensor
 DIMMER=Dimmer
 HANDLE=Griffkontakt
-MOTION_DETECTOR=Bewegungssensor
+MOTION_DETECTOR=Bewegungssensorlanguages_de_DE.properties
```
Do you really want to add the "languages_de_DE.properties" suffix? ;)

Zukye replied:
Well, it was cherry-picked from https://github.com/openbase/bco.bcozy/commit/35a432e83eb3bbd16b6fabaa66e3db4fb07d8575 ;)

DivineThreepwood commented on src/main/java/org/openbase/bco/bcozy/model/UserData.java:
```diff
@@ -0,0 +1,201 @@
+    }
+
+    @Override
+    public int hashCode() {
```
This can do the HashCodeBuilder (Apache Commons Lang 3.6 API) for you:
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/HashCodeBuilder.html

Zukye replied:
Replaced with Objects.hashCode, which is even simpler, hope this is ok?

DivineThreepwood commented on src/main/java/org/openbase/bco/bcozy/model/UserData.java:
```diff
@@ -0,0 +1,201 @@
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
This can do the EqualsBuilder (Apache Commons Lang 3.6 API) for you:
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/EqualsBuilder.html

Zukye replied:
Replaced with Objects.equals, which is even simpler, hope this is ok?

DivineThreepwood replied:
Seems to be ok, i just fixed your class type check.

DivineThreepwood reviewed (changes requested):
Some minor changes required.

Commit:
Fix language_de_DE, refs #53

Commit:
Remove unnecessary important from css, refs #53

Commit:
Replace handwritten equals/hashcode with method from Objects, refs #53

Commit:
Fix Language-Encoding, refs #53

DivineThreepwood committed:
Update UserData.java

Replaced ``if (o == null || getClass() != o.getClass())`` by ``if (!(o instanceof UserData))`` to make equals check stable against inheritance.
The null check is there already included.

DivineThreepwood committed:
Update UserSettingsController.java

Exception handling fixed.

DivineThreepwood reviewed (approved):
ok
